uptrain/operators/language/generation.py

```diff
@@ -170,6 +171,104 @@ def run(self, data: pl.DataFrame) -> TYPE_TABLE_OUTPUT:
         }
 
 
+@register_op
+class TopicGenerator(ColumnOp):
+    """
+    Takes a table of clustered texts and identifies the topic for each cluster 
+
+    Attributes:
+        col_in_cluster_index (str): The name of the column containing the cluster index.
+        col_in_embs (str): The name of the column containing the embeddings.
+        col_in_text (str): The name of the column containing the text where the grouping needs to be performed.
+        col_out_text (str): The name of the column containing the topic for each entry.
+        temperature (float): Temperature for the LLM to generate responses.
+        model (str): which LLM to use for topic generator.
+
+    Returns:
+        TYPE_TABLE_OUTPUT: A dictionary containing the dataset with the output text.
+    """
+    col_in_cluster_index: str = 'cluster_index'
```

Comment:
We should restructure it to assume clustering as an optional? Also top5 -> 5 should be a parameter @ashish-1600